tasking/pulp/tasking/celery_app.py

```diff
@@ -69,6 +72,51 @@ def initialize_worker(sender, instance, **kwargs):
     # Delete any potential old state
     delete_worker(sender, normal_shutdown=True)
 
+    if sender.startswith(constants.RESOURCE_MANAGER_WORKER_NAME):
+        get_resource_manager_lock(sender)
+
+
+def get_resource_manager_lock(name):
+    """
+    Tries to acquire the resource manager lock. If the lock cannot be acquired immediately, it
+    will wait until the currently active instance becomes unavailable, at which point the worker
+    cleanup routine will clear the lock for us to acquire. A worker record will be created so that
+    the waiting resource manager will appear in the Status API. We override the SIGTERM signal
+    handler so that that the worker record will be immediately cleaned up if the process is killed
+    while in this states.
+
+    :param name:   The hostname of the worker
+    :type  name:   basestring
+    """
+    assert name.startswith(constants.RESOURCE_MANAGER_WORKER_NAME)
+
+    lock = TaskLock(name=name, lock=TaskLock.RESOURCE_MANAGER)
+    worker, created = Worker.objects.get_or_create(name=name)
+
+    with custom_sigterm_handler(name):
+        # Whether this is the first lock availability check for this instance
+        _first_check = True
+
+        while True:
+            worker.heartbeat()
+
+            try:
+                lock.save()
+
+                msg = _("Resource manager '%s' has acquired the resource manager lock" % name)
```

Comment:
Can these L#106 - L#108 be moved to an `else:` clause since we aren't interested in the `except` clause applying to it.